### PR TITLE
OpenMesh: processor for computing vertex normals

### PR DESCRIPTION
**Pull request by martinfalk**

### CODE REVIEW

File: misc/openmesh/src/processors/vertexnormals.cpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+#include <modules/openmesh/processors/vertexnormals.h>
+#include <inviwo/core/datastructures/geometry/mesh.h>
+#include <modules/openmesh/utils/openmeshconverters.h>
+
+namespace inviwo {
+
+// The Class Identifier has to be globally unique. Use a reverse DNS naming scheme
+const ProcessorInfo VertexNormals::processorInfo_{
+    "org.inviwo.VertexNormals",  // Class identifier
```
Comment (ResearchDaniel) [inline misc/openmesh/src/processors/vertexnormals.cpp]: Might be a name clash if someone decides to implement it without openmesh?
Perhaps we should have some convention for these external modules?

File: misc/openmesh/include/modules/openmesh/processors/vertexnormals.h [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+#include <inviwo/core/common/inviwo.h>
+#include <inviwo/core/processors/processor.h>
+#include <inviwo/core/ports/meshport.h>
+#include <inviwo/core/properties/boolproperty.h>
+
+namespace inviwo {
+
+/** \docpage{org.inviwo.VertexNormals, Vertex Normals}
+ * ![](org.inviwo.VertexNormals.png?classIdentifier=org.inviwo.VertexNormals)
+ * generates vertex normals for the input mesh. Existing vertex normals will only be overwritten if
```
Comment (ResearchDaniel) [inline misc/openmesh/include/modules/openmesh/processors/vertexnormals.h]: Are there any restrictions on the input mesh?

File: misc/openmesh/src/processors/vertexnormals.cpp (hunk 1 above)
Comment (r-englund) [reply misc/openmesh/src/processors/vertexnormals.cpp]: Myabe `org.inviwo.oom.VertexNormals`?

File: misc/openmesh/include/modules/openmesh/processors/vertexnormals.h (hunk 2 above)
Comment (r-englund) [reply misc/openmesh/include/modules/openmesh/processors/vertexnormals.h]: If it doesn't have any triangles, nothing will happen. 
And any point- or line index buffers are lost in the conversion to `OOM::Mesh`

File: misc/openmesh/src/processors/vertexnormals.cpp [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        outport_.setData(inport_.getData());
+        return;
+    } 
+
+    auto mesh = openmeshutil::fromInviwo(*inport_.getData(),
+                                         openmeshutil::TransformCoordinates::DataToModel);
+
+    //if (mesh.has_vertex_normals()) {
+    //    // get rid of existing normals
+    //    mesh.release_vertex_normals();
```
Comment (r-englund) [inline misc/openmesh/src/processors/vertexnormals.cpp]: Remove if not used

File: misc/openmesh/src/processors/vertexnormals.cpp (hunk 1 above)
Comment (martinfalk) [reply misc/openmesh/src/processors/vertexnormals.cpp]: Good point, settling for now on `"org.inviwo.openmesh.VertexNormals"`

**Pull request merged**